docs/docs/gatsby-magic.md

```diff
@@ -0,0 +1,88 @@
+
+Themes for various use cases are added to the Gatsby ecosystem daily, such as documentation themes, UI library implementations, and more. It’s also possible to create your own theme and either distribute it for people in the community to use or keep it for internal use.
+
+To learn more about Gatsby themes including themes available in the plugin library, check out the [themes section](https://www.gatsbyjs.org/docs/themes/) of the docs.
+
+## Routing
+
+Gatsby sites are full React web applications, meaning they have access to client-side APIs including routing (a fancy word for matching a URL with a part of a web page using JavaScript). Gatsby creates paths to access content, handling dynamic [routing](https://www.gatsbyjs.org/docs/glossary#routing) for you and providing performance benefits through [content prefetching](https://www.gatsbyjs.org/docs/routing/#performance-and-prefetching). A lot of the work is done at build time for a site using mostly [static content](https://www.gatsbyjs.org/docs/adding-app-and-website-functionality/#static-pages). To handle applications that include functionality that cannot be handled at build time or through [rehydration](https://www.gatsbyjs.org/docs/adding-app-and-website-functionality/#how-hydration-makes-apps-possible), including authentication or retrieving dynamic content, you can use [client-only routes](https://www.gatsbyjs.org/docs/client-only-routes-and-user-authentication) using [@reach/router](https://www.gatsbyjs.org/docs/reach-router-and-gatsby/) which is built into Gatsby.
+
+Gatsby makes it easy to create HTML pages that hook into its routing system. Here are some examples of [how to create pages](https://www.gatsbyjs.org/docs/creating-and-modifying-pages) in Gatsby:
+
+- In your site’s gatsby-node.js by implementing the API [createPages](https://www.gatsbyjs.org/docs/node-apis/#createPages)
+- Gatsby core automatically turns React components in src/pages into pages
+- Plugins can also implement createPages and create pages for you
+
+A big part of how Gatsby builds work is its compilation of static HTML pages that are initially fetched from a server and then rehydrated into a full React web application when browser JavaScript takes over. This is where client-side routing comes into the picture with @reach/router, as that’s how Gatsby links pages together.
+
+Routes in Gatsby are generated for you under the hood by creating pages. Changing a route would depend on how that page was created: either changing a filename in `src/pages` or updating a slug in a post’s frontmatter metdata for a dynamic blog route.
```

Comment:
Should this say file path? Since it is a combination of the file name itself and the directories it's in? Or is that over-complicating things?